README.md

```diff
@@ -74,5 +75,6 @@ Our dependencies are (see [requirements.txt](requirements.txt)):
 | Component | About | License |
 | --- | --- | --- |
 | numpy | Array processing library | BSD 3-Clause |
+| einops | Tensor processing utilities | MIT |
```

Comment:
How much will einops give us? It looks very nice, but I'm generally very cautious of adding library dependencies to libraries (much more bullish on applications). If it gives nicer error messages when we don't do explicit shape checking elsewhere, you've won me over!

TBH as I write this, I gave it a quick check, and I think I'm convinced!

```
>>> einops.rearrange(np.ones((2,3,4,5)), "a b c -> c a b").shape
...
einops.EinopsError:  Error while processing rearrange-reduction pattern "a b c -> c a b".
 Input tensor shape: (2, 3, 4, 5). Additional info: {}.
 Expected 3 dimensions, got 4

# vs

>>> np.ones((2,3,4,5)).transpose(2, 0, 1).shape
...
ValueError: axes don't match array
```

Reply:
Nicer error messages for sure!

It's also so much easier to read, write, and reason about tensor shapes and layouts this way! Given poptorch is very opaque about how it represents data across devices to users I think this is worthwhile.

Reply:
:+1: agree - I don't think this dependency will cause issues & will bring substantial benefit.